### PR TITLE
AAE-8904 filter groups by application keycloak

### PR DESCRIPTION
**Pull request by Giovanni007**
Filter groups by application + some refactoring removing roles from /groups and /users API response
Ref: https://github.com/Activiti/Activiti/issues/3954

### CODE REVIEW

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java [hunk 1]
```diff
@@ -72,32 +73,44 @@ public List<User> findUsers(UserSearchParams userSearchParams) {
...
         } else {
-            return filterUsers(users, userSearchParams);
+            return filterUsersInRealmScope(users, userSearchParams);
         }
     }
 
-    private List<User> filterUsers(List<User> users ,UserSearchParams userSearchParams) {
+    private List<User> filterUsersInRealmScope(List<User> users ,UserSearchParams userSearchParams) {
+        Map<String, List<Role>> usersRolesMapping = new HashMap<>();
+        if(!CollectionUtils.isEmpty(userSearchParams.getRoles())) {
```
Comment (mauriziovitale) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java:82]: can't we wrap this logic into a function? mapUserWithRealmRole? enrichUserWithRole?

Comment (Giovanni007) [reply activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java:82]: sure, good point. I'll do thanks

**Review by jesty — APPROVED** [review]
LGTM

**Review by mauriziovitale — APPROVED** [review]

**Comment by oliveralfresco2018** [comment]
@Giovanni007 Does this need adding to the release notes?
If so can we have a chat about what this actually means?
thanks.

**Pull request merged**